src/regressionmodel.jl

```diff
@@ -34,19 +34,25 @@ Return the mean of the response.
 function meanresponse end
 
 """
-    modelmatrix(model::RegressionModel)
+    modelmatrix(model::RegressionModel; weighted::Bool=false)
 
-Return the model matrix (a.k.a. the design matrix).
+Return the model matrix (design matrix) or, if `weighted=true` the weighted 
+model matrix, i.e. `X * sqrt.(W)`, where `X` is the model matrix and
+`W` is the diagonal matrix whose elements are the [model weights](@ref weights(::StatisticalModel)). 
 """
-function modelmatrix end
+function modelmatrix(model::RegressionModel; weighted::Bool=false) end
```

Comment:
Adding arguments means that you define a method which returns `nothing`, which isn't what we want here.
```suggestion
function modelmatrix end
```